### PR TITLE
Christine Movius Solution

### PR DESCRIPTION
**Pull request by christine-sm**
Comfort: 4
Completion: 5

A user can successfully view, add, delete, and edit to-do's. Becky and I played around with using a `data-ng-model` to edit the data, but Sanjaya discovered an Angular out of the box attribute called `contenteditable` that can be used to easily edit the string.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,12 +1,48 @@
...
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css" integrity="sha384-1q8mTJOASx8j1Au+a5WDVnPi2lkFfwwEAa8hDDdjZlpLegxhjVME1fgjWPGmkzs7" crossorigin="anonymous">
+    <title>Angular ToDo App</title>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>
+    <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.5.2/angular.min.js"></script>
+    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/js/bootstrap.min.js" integrity="sha384-0mSbJDEHialfmuBBQP6A4Qrprq5OVfW37PRR3j5ELqxss1yVqOtnepnHVP9aJ7xS" crossorigin="anonymous"></script>
+    <script src="data.js"></script>
+    <script src="xeditable.min.js"></script>
```
Comment (beckybeauchamp1) [inline index.html:13]: This was giving me an error. You have a script for it but no `xeditable.min.js` file. It either needs to be added locally or you should use a CDN.

File: app.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "xeditable"
+  ])
+  .controller("todo_controller", TodoController);
+
+  function TodoController() {
+    var vm = this;
+    vm.data = data;
+
+    vm.destroy = function(todo){
+      var index = vm.data.indexOf(todo);
```
Comment (beckybeauchamp1) [inline app.js:15]: Nice job with this destroy method!

File: index.html [hunk 3]
```diff
@@ -1,12 +1,48 @@
...
+    <main data-ng-controller="todo_controller as vm">
+      <table class="table table-striped">
+        <thead>
+          <tr>
+            <td></td>
+          </tr>
+        </thead>
+        <tbody>
+          <tr data-ng-repeat="todo in vm.data">
+            <td>{{$index + 1}}</td>
```
Comment (beckybeauchamp1) [inline index.html:30]: You could also write a create method like this: 

```
  <td><input type="text" data-ng-model="newTask"></td>
  <td><button data-ng-click="vm.createTodo({'task': newTask})">SUBMIT</button></td>
```

File: app.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      vm.data.splice(index, 1);
+    };
+
+    vm.new_todo = {};
+    vm.create = function(){
```
Comment (beckybeauchamp1) [inline app.js:20]: Then, write a create method like this: 

```
    vm.createTodo = function(todo){
      vm.data.push(todo);
    };
```

Comment (beckybeauchamp1) [inline app.js:20]: It's totally fine the way you have it now! But hopefully this at least gives you another idea or way to solve it

File: index.html (hunk 1 above)
Comment (christine-sm) [inline index.html:13]: Ahh I previously had the `xeditable.min.js` file in here which provides an editable form function but then I deleted it when I no longer needed it. However I forgot to delete the link to the file!

File: index.html [hunk 5]
```diff
@@ -1,12 +1,48 @@
...
+            <td contenteditable="true">{{todo.task}}</td>
+            <!-- <td><input type="text" data-ng-model="todo.task"></td> -->
+            <!--  Thanks for your help Becky! Shortly after our one on one we discovered an out of the box Angular attribute called contenteditable. It's awesome!-->
+            <td data-ng-click="vm.destroy(todo)">&cross;</td>
+          </tr>
+          <tr>
+            <td>Add!</td>
+            <td>
+              <input type="text" data-ng-model="vm.new_todo.task" placeholder="To-Do" />
```
Comment (beckybeauchamp1) [inline index.html:39]: nice job here!

File: index.html (hunk 1 above)
Comment (beckybeauchamp1) [inline index.html:13]: No worries, it was just giving me errors :)

**Comment by beckybeauchamp1** [comment]
Hi Christine,

Great job on this assignment! Please let me know if you have any further questions :+1:

**Pull request closed without merging**